src/trs/hmtrs.rs

```diff
@@ -34,144 +92,106 @@ pub struct HMTRS {
     pub ctx: Context,
 }
 impl HMTRS {
-    /// the size of the HMTRS (the sum over the size of the rules in the [`TRS`])
+    /// The size of the HMTRS (the sum over the size of the rules in the [`TRS`])
     ///
     /// [`TRS`]: ../../term_rewriting/struct.TRS.html
     pub fn size(&self) -> usize {
         self.trs.size()
     }
-    /// Convert the `HMTRS` to a `String`.
-    pub fn display(&self) -> String {
-        self.trs.display(&self.signature)
-    }
-    /// All the free variables in the lexicon.
-    pub fn free_vars(&self) -> Vec<pVar> {
+    /// All the free type variables in the lexicon.
+    pub fn free_vars(&self) -> Vec<TypeVar> {
         let vars_fvs = self.vars.iter().flat_map(|x| x.free_vars());
         let ops_fvs = self.ops.iter().flat_map(|x| x.free_vars());
         vars_fvs.chain(ops_fvs).unique().collect()
     }
-    /// Infer the `Type` of a `Variable`.
-    ///
-    /// No [`Context`] is necessary for inference here; `self` either knows the correct [`TypeSchema`] or generates a [`TypeError`].
-    ///
-    /// [`Context`]: ../../polytype/struct.Context.html
-    /// [`TypeSchema`]: ../../polytype/struct.TypeSchema.html
-    /// [`TypeError`]: struct.TypeError.html
```

Comment:
Why delete these docs?

Reply:
The diff you were looking at only had me changing the docs, not removing them. However, I ended up removing exposure (non-`pub`) on these two functions in favor of `infer_term` and the (to-be-added) `infer_atom` (which will pretty much replace `fast_update`).